doc/specs/stdlib_bitsets.md

```diff
@@ -0,0 +1,2006 @@
+
+#### Class
+
+Elemental function.
+
+#### Argument
+
+`self`: shall be a scalar expression of class `bitset_type`. It is an
+`intent(in)` argument.
+
+#### Result character
+
+The result is an integer scalar of kind `bits_kind`.
+
+#### Result value
+
+The result is the number of defined bits in `self`.
+
+#### Example
+
+```fortran
+    program demo_bits
+        use stdlib_bitsets
+        character(*), parameter :: &
+            bits_0 = '0000000000000000000'
+        type(bitset_64) :: set0
+        call set0 % from_string( bits_0 )
+        if ( set0 % bits() == 19 ) then
+            write(*,*) "FROM_STRING interpreted " // &
+                "BITS_0's size properly."
+        end if
+    end program demo_bits
+```
+
+### `clear` - clears a sequence of one or more bits.
+
+#### Status
+
+Experimental
+
+#### Description
+
+* If only `pos` is present, clears the bit with position `pos` in
+`self`.
+
+* If `start_pos` and `end_pos` are present with `end_pos >= start_pos`
+clears the bits with positions from `start_pos` to `end_pos` in `self`.
+
+* if `start_pos` and `end_pos` are present with `end_pos < start_pos`
+`self` is unmodified.
+
+Note: Positions outside the range 0 to `BITS(SET) -1` are ignored.
+
+#### Syntax
+
+`call self % [[bitset_type(class):clear(bound)]](pos)'
+
+or
+
+`call self % [[bitset_type(class):clear(bound)]](start_pos, end_pos)`
+
+#### Class
+
+Elemental subroutine
+
+#### Arguments
+
+`self`: shall be a scalar variable of class `bitset_type`. It is an
+  `intent(inout)` argument.
+
+`pos`: shall be a scalar integer expression of kind `bits_kind`. It is
+an `intent(in)` argument.
+
+`start_pos`: shall be a scalar integer expression of kind
+`bits_kind`. It is an `intent(in)` argument.
+
+`end_pos`: shall be a scalar integer expression of kind
+`bits_kind`. It is an `intent(in)` argument.
+
+#### Example
+
+```fortran
+    program demo_clear
+        use stdlib_bitsets
+        type(bitset_large) :: set0
+        call set0 % init(166)
+        call set0 % not()
+        if ( set0 % all() ) write(*,*) 'SET0 is properly initialized.'
+        call set0 % clear(165)
+        if ( .not. set0 % test(165) ) write(*,*) 'Bit 165 is cleared.'
+        call set0 % clear(0,164)
+        if ( set0 % none() ) write(*,*) 'All bits are cleared.'
+    end program demo_clear
+```
+
+### `extract` - create a new bitset from a range in an old bitset
+
+#### Status
+
+Experimental
+
+#### Description
+
+Creates a new bitset, `new`, from a range, `start_pos` to `stop_pos`,
+in bitset `old`. If `start_pos` is greater than `stop_pos` the new
+bitset is empty. If `start_pos` is less than zero or `stop_pos` is
+greater than `bits(old)-1` then if `status` is present it has the
+value `index_invalid_error`, otherwise processing stops with an
+informative message.
+
+#### Syntax
+
+`call [[stdlib_bitsets(module):extract(interface)]](new, old, start_pos, stop_pos, status )`
+
+#### Class
+
+Subroutine
```

Comment:
Why is this a subroutine instead a function, e.g.:
`new = extract(old, start_pos,...)`?

Note: I am fine (and even prefer it) as a subroutine. It is mainly to document the PR.

Reply:
The use of a subroutine:
1. Makes it easier to enforce that `new` must be the same type as `old`
2. Allows me to return a status flag, which is useful now, and necessary if I ever want to allow `new` and `old` to be mixed types